scapy/packet.py

```diff
@@ -240,7 +240,7 @@ def prepare_cached_fields(self, flist):
 
             # Remember references
             if isinstance(f.default, (list, dict, set)) or \
-                    isinstance(f.default, RandField):
+                    isinstance(f.default, (RandField, Packet)):
```

Comment:
MMhhh this should be enough, or am I missing something? (not introduced by this commit, but could be fixed here)
```Python
             if isinstance(f.default, (list, dict, set, RandField, Packet)):
```